src/core_plugins/kibana/public/discover/embeddable/search_embeddable_handler.js

```diff
@@ -0,0 +1,77 @@
+import searchTemplate from './search_template.html';
+import angular from 'angular';
+import * as columnActions from 'ui/doc_table/actions/columns';
+import { getPersistedStateId } from 'plugins/kibana/dashboard/panel/panel_state';
+import { EmbeddableHandler } from 'ui/embeddable';
+
+
+export class SearchEmbeddableHandler extends EmbeddableHandler {
+
+  constructor($compile, $rootScope, searchLoader) {
+    super();
+    this.$compile = $compile;
+    this.searchLoader = searchLoader;
+    this.$rootScope = $rootScope;
+    this.name = 'search';
+  }
+
+  async getEditPath(panelId) {
+    return await this.searchLoader.urlFor(panelId);
+  }
+
+  async getTitleFor(panelId) {
+    const savedObject = await this.searchLoader.get(panelId);
+    return savedObject.title;
+  }
+
+  async render(domNode, panel, container) {
```

Comment:
Unfortunately there are a lot of unexpected and hard-to-debug issues that pop-up when native promises or async/await with Angular. I hate to say it but please convert any `async/await` or native promise usage that is interacting with Angular to use Angular promises.